### DISCUSSION TITLE
Readout of MDF4 Measurement Comments

### DISCUSSION THREAD

**Discussion by ByrBen** [danielhrisca/asammdf#470] [Q&A]
Hello everybody,

I started using asammdf a few weeks ago to automate creation of extracts after a few issues getting everything ready that works now. Now I want to get a list of all measurement comments including time stamp.

First I thought for MDF3 I can get them with iter_get_triggers() but that seams to only yield the measurement triggers (hence the name...duh). I can get the complete measurement comment with the info() or in the headerblock but how about the comments that are created during a measurement to mark specific situations (in Inca via Ctrl + K)? In the publicly available documents about the mdf4 format I also cannot find information how the comments are defined in the file.

Has anyone worked with measurement comments so far?

Thanks
Benni

**Comment by danielhrisca** [comment]
> comments that are created during a measurement to mark specific situations (in Inca via Ctrl + K)?

This will be probably available in the trigger comments

```python
for trigger_info_dict in mdf.iter_get_triggers():
    print(trigger_info_dict["comment"])
```

**Reply by ByrBen** [reply]
The iter_get_triggers() only yields triggers that I added with asammdf. It does not list the comments from the original mdf3 measurement file.

For mdf4 file version the iter_get_triggers() is not a defined attribute ("'MDF4' object has no attribute 'iter_get_triggers')

**Reply by danielhrisca** [reply]
I would need an example file to see how Inca stores those user comments

**Reply by ByrBen** [reply]
I uploaded a short measurement file with the following comments:
![image](https://user-images.githubusercontent.com/77772236/114587765-40b3ad80-9c86-11eb-876d-509018dee981.png)

Hope that helps. Thanks.

[comment.zip](https://github.com/danielhrisca/asammdf/files/6305267/comment.zip)

**Reply by danielhrisca** [reply]
In the MDF version 4 you have the ``events`` attribute:

```python
from asammdf import MDF

mdf = MDF(r'comment.mf4')
for event in mdf.events:
    print(event.comment)
```
